dc-cudami-server/dc-cudami-server-webapp/src/main/java/de/digitalcollections/cudami/server/controller/identifiable/agent/FamilyNameController.java

```diff
@@ -62,17 +63,25 @@ public PageResponse<FamilyName> find(
 
   @Operation(summary = "Get a familyname by namespace and id")
   @GetMapping(
-      value = {
-        "/v5/familynames/identifier/{namespace}:{id}",
-        "/v5/familynames/identifier/{namespace}:{id}.json"
-      },
+      value = {"/v5/familynames/identifier/**"},
       produces = MediaType.APPLICATION_JSON_VALUE)
-  public ResponseEntity<FamilyName> getByIdentifier(
-      @PathVariable String namespace, @PathVariable String id) throws IdentifiableServiceException {
+  public ResponseEntity<FamilyName> getByIdentifier(HttpServletRequest request)
+      throws IdentifiableServiceException {
+    String paramString =
+        request.getRequestURI().replaceFirst("^.*?/identifier/", "").replaceFirst("\\.json$", "");
+    if (!paramString.contains(":")) {
+      throw new InvalidParameterException(
+          paramString + " violates expected format <namespace>:<id>");
+    }
+    String[] params = paramString.split(":");
+
+    String namespace = params[0];
+    String id = String.join(":", Arrays.copyOfRange(params, 1, params.length));
```

Comment:
Eigentlich brauchen wir diese Logik doch an etlichen Stellen. Wäre das nicht in einer statischen Methode einer Helperklasse besser aufgehoben?

Reply:
Ist hier herausgenommen und beispielhaft in https://github.com/dbmdz/cudami/pull/1662 umgesetzt